python/mcap/mcap0/data_stream.py

```diff
@@ -1,18 +1,15 @@
 import struct
-from io import BufferedIOBase, BytesIO
+from io import IOBase, BytesIO
 
 from .exceptions import EndOfFile
 from .opcode import Opcode
 
 
 class ReadDataStream:
-    def __init__(self, stream: BufferedIOBase):
+    def __init__(self, stream: IOBase):
         self.__count = 0
         self.__stream = stream
 
-    def __del__(self):
-        self.__stream.close()
```

Comment:
note: `__del__` implementations rarely do what you want them to, since when an object gets deleted is entirely up to the garbage collector. Also, the `__stream` implementation will almost certainly close itself on deletion already.